backend/de.metas.banking/de.metas.banking.camt53/src/main/java/de/metas/banking/camt53/wrapper/AccountStatement2Wrapper.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * #%L
+ * camt53
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.banking.camt53.wrapper;
+
+import ch.qos.logback.classic.Level;
+import de.metas.banking.BankAccount;
+import de.metas.banking.BankAccountId;
+import de.metas.banking.BankId;
+import de.metas.banking.api.BankAccountService;
+import de.metas.banking.camt53.jaxb.camt053_001_02.AccountStatement2;
+import de.metas.banking.camt53.jaxb.camt053_001_02.BalanceType12Code;
+import de.metas.banking.camt53.jaxb.camt053_001_02.CashBalance3;
+import de.metas.banking.camt53.jaxb.camt053_001_02.GenericAccountIdentification1;
+import de.metas.banking.service.BankStatementCreateRequest;
+import de.metas.currency.Currency;
+import de.metas.currency.CurrencyCode;
+import de.metas.currency.CurrencyRepository;
+import de.metas.logging.LogManager;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.OrgId;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.compiere.util.TimeUtil;
+import org.slf4j.Logger;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+import static de.metas.banking.camt53.jaxb.camt053_001_02.CreditDebitCode.CRDT;
+
+@Value
```

Comment:
why `@Value`?

Reply:
btw, what's the purpose of `AccountStatement2Wrapper`?

reading the code, I understand this is nothing more than a BankStatementCreateRequest factory. Isn't it?
If that's then we shall name it accordingly.... or if that sounds/feels ugly maybe how the code is organized might be wrong...

(loud thinking)

Reply:
I used @Value mainly for immutability
Refactored the wrapper logic